CloudVision_Studios/RCF Studio/RCF v4.2.yaml

```diff
@@ -0,0 +1,178 @@
+- service: arista.studio.v1.StudioConfigService
+  method: Set
+  body:
+    value:
+      key:
+        studio_id: 7a161847-e603-49e5-904d-dc02fbef0d4e
+        workspace_id: ''
+      display_name: RCF v4.2
+      description: Code Units Management Tool
+      template:
+        type: TEMPLATE_TYPE_MAKO
+        body: |+
+          <%
+          # If the developer wants to see the container details of the script. 
+          # Defult debug setting is False. The info can be seen in `view build details` after clicking `review Workspace`
+          debug = False
+
+          def output(content: str, indent=3, level=0):
+              '''
+              return: format multiline strings
+              '''
+              contentList = content.splitlines()
+              res = ''
+              for l in contentList:
+                  indentation = ' ' * (indent * level)
+                  newL = indentation + l + '\n'
+                  res += newL
+              return res.rstrip()
+
+          codeUnitsData = {}
+          for codeUnit in codeUnits:
+              codeUnitsData.update({codeUnit['unitName']:output(codeUnit['unitContent'], level=2)})
+
+          deviceToUnits = {}
+          for assign in deviceAssignment.inputs:
+              codeUnitsList = assign['inputs']['value']['assignedUnits']
+              device_id = assign['devices'][0]
+              deviceToUnits.update({device_id:codeUnitsList})
+          %>
+
+          %if debug:
+          ! ${codeUnitsData}
+          ! ${deviceToUnits}
+          %endif
+
+          %if ctx.getDevice().id in deviceToUnits.keys() and len(deviceToUnits[ctx.getDevice().id]) > 0:
```

Comment:
don't need `.keys()`. Python does a search in the keys already when we say `x in y`